official/utils/logs/hooks.py

```diff
@@ -73,14 +73,30 @@ def __init__(self,
     self._total_steps = 0
     self._batch_size = batch_size
     self._warm_steps = warm_steps
+    self._train_op = None
+
+  def specify_train_op(self, train_op):
```

Comment:
I would prefer to define it as a property so that it gets both getter and setter method.